src/main/java/com/mraof/minestuck/world/gen/structure/FrogTemplePiece.java

```diff
@@ -264,38 +263,20 @@ private Selector()
-			{
-				this.next = MSBlocks.GREEN_STONE_BRICK_IGUANA_RIGHT.get().defaultBlockState();
-			} else if(randomBlock == 3)
-			{
-				this.next = MSBlocks.GREEN_STONE_BRICK_NAK_LEFT.get().defaultBlockState();
-			} else if(randomBlock == 2)
-			{
-				this.next = MSBlocks.GREEN_STONE_BRICK_NAK_RIGHT.get().defaultBlockState();
-			} else if(randomBlock == 1)
-			{
-				this.next = MSBlocks.GREEN_STONE_BRICK_SALAMANDER_LEFT.get().defaultBlockState();
-			} else
-			{
-				this.next = MSBlocks.GREEN_STONE_BRICK_SALAMANDER_RIGHT.get().defaultBlockState();
-			}
+			Block[] blockArray = new Block[]{
+					MSBlocks.GREEN_STONE_BRICK_FROG.get(), //only one frog hieroglyph in order to have a component more difficult to find
+					MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(), MSBlocks.GREEN_STONE_BRICK_TURTLE.get(),
+					MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(), MSBlocks.GREEN_STONE_BRICK_SKAIA.get(),
+					MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(), MSBlocks.GREEN_STONE_BRICK_LOTUS.get(),
+					MSBlocks.GREEN_STONE_BRICK_IGUANA_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_LEFT.get(),
+					MSBlocks.GREEN_STONE_BRICK_IGUANA_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_IGUANA_RIGHT.get(),
+					MSBlocks.GREEN_STONE_BRICK_NAK_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_LEFT.get(),
+					MSBlocks.GREEN_STONE_BRICK_NAK_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_NAK_RIGHT.get(),
+					MSBlocks.GREEN_STONE_BRICK_SALAMANDER_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_LEFT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_LEFT.get(),
+					MSBlocks.GREEN_STONE_BRICK_SALAMANDER_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_RIGHT.get(), MSBlocks.GREEN_STONE_BRICK_SALAMANDER_RIGHT.get()
+			};
```

Comment:
Could you turn this into a weighted list instead? See `WeightedRandom` and `WeightedEntry`.

Reply:
It works in game but please check to make sure I used a rational approach